packages/taco/test/conditions/context.test.ts

```diff
@@ -200,6 +200,25 @@ describe('context', () => {
         expect(asObj).toBeDefined();
         expect(asObj[USER_ADDRESS_PARAM]).toBeDefined();
       });
+
+      it.each([0, ''])('accepts on a falsy parameter value: %s', async (falsyParam) => {
```

Comment:
I wasn't sure, but I could argue there may be a contract call returning empty `bytes32`. These parameters are picked by SDK users, so I'm assuming they know what they're doing.

Reply:
Thought about this a bit more.

I don't believe `''` is valid for bytes either sent as a parameter to a contract call or returned from a contract call.

```python
In [23]: w3.is_encodable("bytes", "")
Out[23]: False
```

```python
In [52]: from eth_abi import encode

In [53]: encode(["bytes32"], [b""])
Out[53]: b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'

In [54]: encode(["bytes"], [b""])
Out[54]: b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'

In [55]: encode(["bytes1"], [b""])
Out[55]: b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
```

I believe empty bytes should always be encoded as a hex string with 0s i.e. `0x0000....` and never empty strings. 

```python
In [58]: HexBytes(encode(["bytes32"], [b""])).hex()
Out[58]: '0x0000000000000000000000000000000000000000000000000000000000000000'

In [59]: HexBytes(encode(["bytes"], [b""])).hex()
Out[59]: '0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000'

In [60]: HexBytes(encode(["bytes1"], [b""])).hex()
Out[60]: '0x0000000000000000000000000000000000000000000000000000000000000000'
```

cc @cygnusv , @vzotova who have more expertise in this area.

Reply:
I think `0`, `''` and `[]` are all suspect

Reply:
I think `0` is probably fine, it can be an int or bool, which is allowed to be 0.

But yeah `''`, and `[]` are suspect.

Let's see what @cygnusv and @vzotova think about those...

Reply:
Intuitively, `0` could also be `0x0`, which I think also qualifies as `bytes32`. 

I'm looking for a comprehensive solution and IMHO being permissive here is preferred.